core/include/detray/propagator/propagator.hpp

```diff
@@ -218,12 +231,68 @@ struct propagator {
                 // And check the status
                 propagation._heartbeat &=
                     m_navigator.update(propagation, m_cfg);
+
+#if defined(__NO_DEVICE__)
+                if (propagation.do_debug) {
+                    inspect(propagation);
+                }
+#endif                
             }
         }
 
         // Pass on the whether the propagation was successful
         return propagation._navigation.is_complete();
     }
+
+    template <typename state_t>
+    DETRAY_HOST void inspect(state_t &propagation) {
```

Comment:
If you move that directly into the propagator, can you remove the actor that does the same thing?

Reply:
Done